
02/02/2024 11:23:01 - WARNING - __main__ - Process rank: 0, device: cuda:0, n_gpu: 1distributed training: True, 16-bits training: False
/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/datasets/load.py:2483: FutureWarning: 'use_auth_token' was deprecated in favor of 'token' in version 2.14.0 and will be removed in 3.0.0.
You can remove this warning by passing 'token=<use_auth_token>' instead.
  warnings.warn(
[INFO|configuration_utils.py:667] 2024-02-02 11:23:05,685 >> loading configuration file /data/ai/model/chatglm2-6b/config.json
[INFO|configuration_utils.py:667] 2024-02-02 11:23:05,686 >> loading configuration file /data/ai/model/chatglm2-6b/config.json
[INFO|configuration_utils.py:725] 2024-02-02 11:23:05,687 >> Model config ChatGLMConfig {
  "_name_or_path": "/data/ai/model/chatglm2-6b",
  "add_bias_linear": false,
  "add_qkv_bias": true,
  "apply_query_key_layer_scaling": true,
  "apply_residual_connection_post_layernorm": false,
  "architectures": [
    "ChatGLMModel"
  ],
  "attention_dropout": 0.0,
  "attention_softmax_in_fp32": true,
  "auto_map": {
    "AutoConfig": "configuration_chatglm.ChatGLMConfig",
    "AutoModel": "modeling_chatglm.ChatGLMForConditionalGeneration",
    "AutoModelForCausalLM": "modeling_chatglm.ChatGLMForConditionalGeneration",
    "AutoModelForSeq2SeqLM": "modeling_chatglm.ChatGLMForConditionalGeneration",
    "AutoModelForSequenceClassification": "modeling_chatglm.ChatGLMForSequenceClassification"
  },
  "bias_dropout_fusion": true,
  "classifier_dropout": null,
  "eos_token_id": 2,
  "ffn_hidden_size": 13696,
  "fp32_residual_connection": false,
  "hidden_dropout": 0.0,
  "hidden_size": 4096,
  "kv_channels": 128,
  "layernorm_epsilon": 1e-05,
  "model_type": "chatglm",
  "multi_query_attention": true,
  "multi_query_group_num": 2,
  "num_attention_heads": 32,
  "num_layers": 28,
  "original_rope": true,
  "pad_token_id": 0,
  "padded_vocab_size": 65024,
  "post_layer_norm": true,
  "pre_seq_len": null,
  "prefix_projection": false,
  "quantization_bit": 0,
  "rmsnorm": true,
  "seq_length": 32768,
  "tie_word_embeddings": false,
  "torch_dtype": "float16",
  "transformers_version": "4.30.2",
  "use_cache": true,
  "vocab_size": 65024
}
[INFO|tokenization_utils_base.py:1821] 2024-02-02 11:23:05,688 >> loading file tokenizer.model
[INFO|tokenization_utils_base.py:1821] 2024-02-02 11:23:05,688 >> loading file added_tokens.json
[INFO|tokenization_utils_base.py:1821] 2024-02-02 11:23:05,688 >> loading file special_tokens_map.json
[INFO|tokenization_utils_base.py:1821] 2024-02-02 11:23:05,688 >> loading file tokenizer_config.json
[INFO|modeling_utils.py:2575] 2024-02-02 11:23:05,732 >> loading weights file /data/ai/model/chatglm2-6b/pytorch_model.bin.index.json
[INFO|configuration_utils.py:577] 2024-02-02 11:23:05,733 >> Generate config GenerationConfig {
  "_from_model_config": true,
  "eos_token_id": 2,
  "pad_token_id": 0,
  "transformers_version": "4.30.2"
}


Loading checkpoint shards:  71%|████████████████████████████████████████████████▌                   | 5/7 [00:03<00:01,  1.37it/s]
Loading checkpoint shards: 100%|████████████████████████████████████████████████████████████████████| 7/7 [00:04<00:00,  1.45it/s]
[INFO|modeling_utils.py:3295] 2024-02-02 11:23:10,591 >> All model checkpoint weights were used when initializing ChatGLMForConditionalGeneration.
[WARNING|modeling_utils.py:3297] 2024-02-02 11:23:10,591 >> Some weights of ChatGLMForConditionalGeneration were not initialized from the model checkpoint at /data/ai/model/chatglm2-6b and are newly initialized: ['transformer.prefix_encoder.embedding.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
[INFO|modeling_utils.py:2927] 2024-02-02 11:23:10,592 >> Generation config file not found, using a generation config created from the model config.
prefix:
inputs什么是蚂蚁庄园
input_ids [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 64790, 64792, 30910, 35318, 41069, 46980]
inputs 什么是蚂蚁庄园
label_ids [64790, 64792, 30910, 41069, 46980, 41680, 33053, 39063, 32488, 39330, 38346, 35439, 32033, 31155]
labels 蚂蚁庄园是一款基于支付宝支付行为的农场养成游戏。
Running tokenizer on validation dataset: 100%|█████████████████████████████████████████| 315/315 [00:00<00:00, 8032.82 examples/s]
02/02/2024 11:23:14 - INFO - __main__ - *** Evaluate ***
[INFO|trainer.py:577] 2024-02-02 11:23:14,104 >> max_steps is given, it will override any value given in num_train_epochs
[INFO|trainer.py:3200] 2024-02-02 11:23:14,107 >> ***** Running Evaluation *****
[INFO|trainer.py:3202] 2024-02-02 11:23:14,107 >>   Num examples = 315
[INFO|trainer.py:3205] 2024-02-02 11:23:14,107 >>   Batch size = 4
[INFO|configuration_utils.py:577] 2024-02-02 11:23:14,117 >> Generate config GenerationConfig {
  "_from_model_config": true,
  "eos_token_id": 2,
  "pad_token_id": 0,
  "transformers_version": "4.30.2"
}
  0%|                                                                                                      | 0/79 [00:00<?, ?it/s][INFO|configuration_utils.py:577] 2024-02-02 11:23:22,065 >> Generate config GenerationConfig {
  "_from_model_config": true,
  "eos_token_id": 2,
  "pad_token_id": 0,
  "transformers_version": "4.30.2"
}
  3%|██▍                                                                                           | 2/79 [00:13<08:24,  6.55s/it][INFO|configuration_utils.py:577] 2024-02-02 11:23:35,174 >> Generate config GenerationConfig {
  "_from_model_config": true,
  "eos_token_id": 2,
  "pad_token_id": 0,
  "transformers_version": "4.30.2"
}
  4%|███▌                                                                                          | 3/79 [00:27<12:13,  9.66s/it][INFO|configuration_utils.py:577] 2024-02-02 11:23:49,171 >> Generate config GenerationConfig {
  "_from_model_config": true,
  "eos_token_id": 2,
  "pad_token_id": 0,
  "transformers_version": "4.30.2"
}
  5%|████▊                                                                                         | 4/79 [00:48<17:32, 14.04s/it][INFO|configuration_utils.py:577] 2024-02-02 11:24:10,570 >> Generate config GenerationConfig {
  "_from_model_config": true,
  "eos_token_id": 2,
  "pad_token_id": 0,
  "transformers_version": "4.30.2"
}
  6%|█████▉                                                                                        | 5/79 [00:55<14:27, 11.72s/it][INFO|configuration_utils.py:577] 2024-02-02 11:24:17,937 >> Generate config GenerationConfig {
  "_from_model_config": true,
  "eos_token_id": 2,
  "pad_token_id": 0,
  "transformers_version": "4.30.2"
}
Error in sys.excepthook:
Traceback (most recent call last):
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/linecache.py", line 46, in getlines
    return updatecache(filename, module_globals)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/linecache.py", line 136, in updatecache
    with tokenize.open(fullname) as fp:
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/tokenize.py", line 396, in open
    encoding, lines = detect_encoding(buffer.readline)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/tokenize.py", line 365, in detect_encoding
    first = read_or_stop()
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/tokenize.py", line 323, in read_or_stop
    return readline()
KeyboardInterrupt
Original exception was:
Traceback (most recent call last):
  File "/data/ai/ChatGLM2-6B/ptuning/main.py", line 480, in <module>
    main()
  File "/data/ai/ChatGLM2-6B/ptuning/main.py", line 413, in main
    metrics = trainer.evaluate(metric_key_prefix="eval", do_sample=True, top_p=0.7, max_length=max_seq_length, temperature=0.95)
  File "/data/ai/ChatGLM2-6B/ptuning/trainer_seq2seq.py", line 79, in evaluate
    return super().evaluate(eval_dataset, ignore_keys=ignore_keys, metric_key_prefix=metric_key_prefix)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/transformers/trainer.py", line 3053, in evaluate
    output = eval_loop(
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/transformers/trainer.py", line 3245, in evaluation_loop
    loss, logits, labels = self.prediction_step(model, inputs, prediction_loss_only, ignore_keys=ignore_keys)
  File "/data/ai/ChatGLM2-6B/ptuning/trainer_seq2seq.py", line 203, in prediction_step
    generated_tokens = self.model.generate(**gen_kwargs)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/transformers/generation/utils.py", line 1572, in generate
    return self.sample(
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/transformers/generation/utils.py", line 2619, in sample
    outputs = self(
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/ubuntu/.cache/huggingface/modules/transformers_modules/chatglm2-6b/modeling_chatglm.py", line 931, in forward
    transformer_outputs = self.transformer(
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/ubuntu/.cache/huggingface/modules/transformers_modules/chatglm2-6b/modeling_chatglm.py", line 824, in forward
    hidden_states, presents, all_hidden_states, all_self_attentions = self.encoder(
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/ubuntu/.cache/huggingface/modules/transformers_modules/chatglm2-6b/modeling_chatglm.py", line 634, in forward
    layer_ret = layer(
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/ubuntu/.cache/huggingface/modules/transformers_modules/chatglm2-6b/modeling_chatglm.py", line 538, in forward
    attention_output, kv_cache = self.self_attention(
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/ubuntu/.cache/huggingface/modules/transformers_modules/chatglm2-6b/modeling_chatglm.py", line 404, in forward
    query_layer = apply_rotary_pos_emb(query_layer, rotary_pos_emb)
KeyboardInterrupt